addons/account/migrations/13.0.1.1/post-migration.py

```diff
@@ -292,6 +292,21 @@ def migration_invoice_moves(env):
         )
         if not env.cr.fetchone():
             break  # exit condition not having more duplicates
+    # Checking the missing lines to import
+    openupgrade.logged_query(
+        env.cr, """
+        UPDATE account_invoice_line ail
+        SET aml_matched = FALSE
+        """
+    )
+    openupgrade.logged_query(
+        env.cr, """
+        UPDATE account_invoice_line ail
+        SET aml_matched = TRUE
+        FROM account_move_line aml
+        WHERE aml.old_invoice_line_id = ail.id
+        """
+    )
```

Comment:
```suggestion
```
I think this part is not needed, because aml_matched already have those values there

Reply:
I thought the same, but when I tested, I found it was necessary as the not related lines were marked as aml_matched

Reply:
I don't understand. the aml_matched is filled previously with for the invoice lines matching move lines.

Reply:
@MiquelRForgeFlow I have serious doubts on this aml_matched code. I got 2 cases now where it just isn't working. It basically just displays the same problem as before except instead of first it is last match. @etobella is actually just covering the effect, not the cause (which IMHO is the total ignorance of line values in the algorithm), but I spent so long inside that while loop I just can't work out where it goes wrong. It looks perfect.

I honestly feel like we have adopted the wrong approach here altogether with invoice migration and that Odoo SA has got it more right by applying ever restrictive rules and making decision based on amounts, among other things. 

I used this really ugly code here (note it doesn't do grouped, because I didn't need) https://github.com/odoonz/OpenUpgrade/blob/13.0/addons/account/migrations/13.0.1.1/post-migration.py to kind of emulate and mismatches dropped from 300,000+ with current code to 57. Of the 57, 51 were due to an import/export bug allowing overwriting posted journals.

Now Odoo SA does make grouped invoice lines into comments and in this case I feel we do better. 

FYI, here is some enterprise log output from a currently running enterprise mig, I think all in it applies 250 of these conditions. They are actually stored as data in postgres table so can provide all.
```
2021-09-20 06:50:39,940 2012 INFO db_31437 odoo.addons.base.maintenance.migrations.account.saas-12.4.end-09-account-pocalypse: insert query 183: cond:{'same_account', 'same_trimmed_name', 'paid_invoice_posted_move', 'nearly_same_amount', 'same_product'} 
2021-09-20 06:50:43,403 2012 INFO db_31437 odoo.addons.base.maintenance.migrations.account.saas-12.4.end-09-account-pocalypse: invoices: fill line mapping with +8/-8 entries using conditions {'same_account', 'same_trimmed_name', 'paid_invoice_posted_move', 'nearly_same_amount', 'same_product'} 
2021-09-20 06:50:43,754 2012 INFO db_31437 odoo.addons.base.maintenance.migrations.account.saas-12.4.end-09-account-pocalypse: invoices: still 55318 to match 
2021-09-20 06:50:43,754 2012 INFO db_31437 odoo.addons.base.maintenance.migrations.account.saas-12.4.end-09-account-pocalypse: insert query 184: cond:{'same_chopped_name', 'same_account', 'paid_invoice_posted_move', 'nearly_same_amount', 'same_product'} 
2021-09-20 06:50:44,196 2012 INFO db_31437 odoo.addons.base.maintenance.migrations.account.saas-12.4.end-09-account-pocalypse: invoices: fill line mapping with +0/-0 entries using conditions {'same_chopped_name', 'same_account', 'paid_invoice_posted_move', 'nearly_same_amount', 'same_product'} 
2021-09-20 06:50:44,543 2012 INFO db_31437 odoo.addons.base.maintenance.migrations.account.saas-12.4.end-09-account-pocalypse: invoices: still 55318 to match 
2021-09-20 06:50:44,543 2012 INFO db_31437 odoo.addons.base.maintenance.migrations.account.saas-12.4.end-09-account-pocalypse: insert query 185: cond:{'same_name_first_line', 'same_account', 'paid_invoice_posted_move', 'nearly_same_amount', 'same_product'} 
2021-09-20 06:50:48,081 2012 INFO db_31437 odoo.addons.base.maintenance.migrations.account.saas-12.4.end-09-account-pocalypse: invoices: fill line mapping with +8/-8 entries using conditions {'same_name_first_line', 'same_account', 'paid_invoice_posted_move', 'nearly_same_amount', 'same_product'} 
2021-09-20 06:50:48,428 2012 INFO db_31437 odoo.addons.base.maintenance.migrations.account.saas-12.4.end-09-account-pocalypse: invoices: still 55318 to match 
2021-09-20 06:50:48,428 2012 INFO db_31437 odoo.addons.base.maintenance.migrations.account.saas-12.4.end-09-account-pocalypse: insert query 186: cond:{'same_account', 'paid_invoice_posted_move', 'nearly_same_amount', 'substring_name', 'same_product'} 
2021-09-20 06:50:51,878 2012 INFO db_31437 odoo.addons.base.maintenance.migrations.account.saas-12.4.end-09-account-pocalypse: invoices: fill line mapping with +8/-8 entries using conditions {'same_account', 'paid_invoice_posted_move', 'nearly_same_amount', 'substring_name', 'same_product'} 
2021-09-20 06:50:52,228 2012 INFO db_31437 odoo.addons.base.maintenance.migrations.account.saas-12.4.end-09-account-pocalypse: invoices: still 55318 to match 
2021-09-20 06:50:52,228 2012 INFO db_31437 odoo.addons.base.maintenance.migrations.account.saas-12.4.end-09-account-pocalypse: insert query 187: cond:{'same_account', 'same_not_null_product', 'paid_invoice_posted_move', 'nearly_same_amount'} 
2021-09-20 06:50:55,712 2012 INFO db_31437 odoo.addons.base.maintenance.migrations.account.saas-12.4.end-09-account-pocalypse: invoices: fill line mapping with +8/-8 entries using conditions {'same_account', 'same_not_null_product', 'paid_invoice_posted_move', 'nearly_same_amount'} 
2021-09-20 06:50:56,067 2012 INFO db_31437 odoo.addons.base.maintenance.migrations.account.saas-12.4.end-09-account-pocalypse: invoices: still 55318 to match 
```

Reply:
@gdgellatly of course Odoo SA migration code might be more convoluted, but here @pedrobaeza advocates a simple readable non-repeatable code.

Reply:
@MiquelRForgeFlow I'm all for simple code. But as I say I got 2 cases where it just produces wrong results. And I still think ignoring the amounts is flawed. Obviously my hacky flattening is not how Odoo SA does it, it is just 1 function called with different arguments.

And it sounds like @etobella is experiencing something related.